subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/AbstractSourceIncrementalCompilationIntegrationTest.groovy

```diff
@@ -1099,26 +1099,29 @@ sourceSets {
 
     def "recompiles all dependents when no jar analysis is present"() {
         given:
+        executer.requireDaemon().requireIsolatedDaemons().requireOwnGradleUserHomeDir()
         source """class A {
-            com.google.common.base.Splitter splitter;
+            org.apache.commons.lang3.StringUtils utils;
         }"""
         source """class B {}"""
 
         buildFile << """
             ${mavenCentralRepository()}
-            dependencies { implementation 'com.google.guava:guava:21.0' }
+            dependencies { implementation 'org.apache.commons:commons-lang3:3.8' }
         """
         outputs.snapshot { succeeds language.compileTaskName }
 
-        when:
-        executer.requireOwnGradleUserHomeDir()
-        source """class B {
-            //some change
-        }"""
+        when: "In-memory and on-disk caches are clean"
+        executer.stop()
+        file("user-home").deleteDir()
+
+        and: "We're no longer using the old dependency"
+        buildFile << "dependencies { implementation 'org.apache.commons:commons-lang3:3.9' }"
```

Comment:
Using this specific dependency, because it has a different hash, but no changes to constants or other full recompile reasons